src/TasksModule/Components/TasksForm/TasksForm.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useEffect, useState } from 'react'
+import styles from './TasksForm.module.css'
+import { useNavigate, useParams } from 'react-router-dom'
+import { useForm } from 'react-hook-form';
+import axios from 'axios';
+import { baseUrl } from '../../../Constants/Components/Urls';
+import { toast } from 'react-toastify';
+import Loader from "../../../SharedMoudule/Components/Loading/Loading";
+
+export default function TasksForm() {
+    const navigate=useNavigate();
+    const {register,handleSubmit,formState:{errors}}=useForm();
+    const token=localStorage.getItem('userToken');
+    const {action}=useParams();
+    const [currentTask,setCurrentTask]=useState(null);
+    const curruntTaskId=localStorage.getItem('curruntTaskId');
+    const [ProjectsList,setProjectsList]=useState(null);
+    const [UsersList,setUsersList]=useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const onSubmitAdd= async(data:any)=>{
+        setIsLoading(true);
+        // console.log(data);
+        try {
+            let response=await axios.post(`${baseUrl}/Task`,data,{headers:{Authorization:token}});
+            console.log('Add task response ',response);
+            toast.success('Task Added Successfuly');
+            navigate('/dashboard/tasks-list');
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            toast.error(error?.response?.data?.message||'Somthing went wrong!');
+        }
+        
+    }
+
+    const onSubmitUpdate= async(data:any)=>{
+        setIsLoading(true);
+        // console.log(data);
+        // {'title':data.title?data.title:currentTask.title,'description':data.description?data.description:currentTask.description}
+        try {
+            let response=await axios.put(`${baseUrl}/Task/${curruntTaskId}`,
+            {"title": data.title?data.title:currentTask?.title,
```

Comment:
consider using the data object directly instead of accessing individual properties.